src/stactools/naip/utils.py

```diff
@@ -91,3 +92,15 @@ def maybe_extract_id_and_date(cog_href: str) -> Optional[Tuple[str, datetime]]:
         return None
     dt = dateutil.parser.isoparse(m.group("dt"))
     return resource_desc, dt
+
+
+def process_resource_desc(resource_desc: str) -> Optional[Tuple[str, datetime]]:
```

Comment:
you have a check if `resource_desc is not None`, so that indicates the parameter signature should be `resource_desc: Optional[str]`

Reply:
I would recommend splitting this method into two methods. It seems be be doing two unreleated things -- stripping the `.tif` extension from the resource_desc and determining the datetime. It's not ideal to have one method that returns a tuple with unrelated values like this.

Reply:
As the method is written, the signature is not `Optional[Tuple[str, datetime]]` but rather `Tuple[Optional[str], Optional[datetime]]`, since you're always returning a Tuple, but the values in it may be None. (but this method should be split up anyway, so it doesn't matter)

Reply:
Thanks for the feedback. 

Adjusted function to only process the resource desc. The code for dt won't be duplicated and can be within stac.py